src/cmd/flux-exec.c

```diff
@@ -218,6 +267,18 @@ void output_cb (flux_subprocess_t *p, const char *stream)
     }
 }
 
+void credit_cb (flux_subprocess_t *p, const char *stream, int bytes)
+{
+    subprocess_update_credits (p, bytes, true);
+    if (started == rank_count) {
+        int min_credits = subprocess_min_credits ();
+        if (min_credits)
+            flux_watcher_start (stdin_w);
+        else
+            flux_watcher_stop (stdin_w);
```

Comment:
I don't _think_ this else clause is needed since this only adds credits, so the current `min_credits` is never reduced over what might have come before when the watcher was started?